implementation/src/main/java/io/smallrye/jwt/auth/principal/DefaultJWTTokenParser.java

```diff
@@ -53,11 +53,22 @@ public JwtContext parse(final String token, final JWTAuthContextInfo authContext
         try {
             JwtConsumerBuilder builder = new JwtConsumerBuilder()
                     .setRequireExpirationTime()
-                    .setRequireSubject()
-                    .setSkipDefaultAudienceValidation()
-                    .setJwsAlgorithmConstraints(
-                            new AlgorithmConstraints(AlgorithmConstraints.ConstraintType.WHITELIST,
-                                    AlgorithmIdentifiers.RSA_USING_SHA256));
+                    .setSkipDefaultAudienceValidation();
+
+            if (authContextInfo.getDefaultSubjectClaim() == null ||
+                    Claims.sub.name().equals(authContextInfo.getDefaultSubjectClaim())) {
```

Comment:
I think we can drop the check for `sub`, it does not save us much

Reply:
Ok